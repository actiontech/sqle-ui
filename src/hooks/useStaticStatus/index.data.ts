import {
  getAuditTaskSQLsV1FilterAuditStatusEnum,
  getAuditTaskSQLsV1FilterExecStatusEnum,
} from '../../api/task/index.enum';
import {
  getWorkflowListV1FilterCurrentStepTypeEnum,
  getWorkflowListV1FilterStatusEnum,
} from '../../api/workflow/index.enum';
import { StaticEnumDictionary } from './index.type';
import { RuleResV1LevelEnum } from '../../api/common.enum';

export const WorkflowStepTypeDictionary: StaticEnumDictionary<getWorkflowListV1FilterCurrentStepTypeEnum> =
  {
    [getWorkflowListV1FilterCurrentStepTypeEnum.sql_review]:
      'order.workflowStatus.review',
    [getWorkflowListV1FilterCurrentStepTypeEnum.sql_execute]:
      'order.workflowStatus.exec',
  };

export const execStatusDictionary: StaticEnumDictionary<getAuditTaskSQLsV1FilterExecStatusEnum> =
  {
    [getAuditTaskSQLsV1FilterExecStatusEnum.initialized]:
      'audit.execStatus.initialized',
    [getAuditTaskSQLsV1FilterExecStatusEnum.doing]: 'audit.execStatus.doing',
    [getAuditTaskSQLsV1FilterExecStatusEnum.failed]: 'audit.execStatus.failed',
    [getAuditTaskSQLsV1FilterExecStatusEnum.succeeded]:
      'audit.execStatus.succeeded',
  };

export const auditStatusDictionary: StaticEnumDictionary<getAuditTaskSQLsV1FilterAuditStatusEnum> =
  {
    [getAuditTaskSQLsV1FilterAuditStatusEnum.initialized]:
      'audit.auditStatus.initialized',
    [getAuditTaskSQLsV1FilterAuditStatusEnum.doing]: 'audit.auditStatus.doing',
    [getAuditTaskSQLsV1FilterAuditStatusEnum.finished]:
      'audit.auditStatus.finished',
  };

export const orderStatusDictionary: StaticEnumDictionary<getWorkflowListV1FilterStatusEnum> =
  {
    [getWorkflowListV1FilterStatusEnum.on_process]: 'order.status.process',
    [getWorkflowListV1FilterStatusEnum.finished]: 'order.status.finished',
    [getWorkflowListV1FilterStatusEnum.canceled]: 'order.status.canceled',
    [getWorkflowListV1FilterStatusEnum.rejected]: 'order.status.reject',
    [getWorkflowListV1FilterStatusEnum.exec_scheduled]:
      'order.status.exec_scheduled',
    [getWorkflowListV1FilterStatusEnum.exec_failed]: 'order.status.exec_failed',
    [getWorkflowListV1FilterStatusEnum.executing]: 'order.status.executing',
  };

export const ruleLevelDictionary: StaticEnumDictionary<RuleResV1LevelEnum> = {
  [RuleResV1LevelEnum.normal]: 'ruleTemplate.ruleLevel.normal',
  [RuleResV1LevelEnum.notice]: 'ruleTemplate.ruleLevel.notice',
  [RuleResV1LevelEnum.warn]: 'ruleTemplate.ruleLevel.warn',
  [RuleResV1LevelEnum.error]: 'ruleTemplate.ruleLevel.error',
};
