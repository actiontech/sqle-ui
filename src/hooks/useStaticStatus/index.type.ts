import { I18nKey } from '../../types/common.type';

export type StaticEnumDictionary<T extends string> = {
  [key in T]: I18nKey;
};
