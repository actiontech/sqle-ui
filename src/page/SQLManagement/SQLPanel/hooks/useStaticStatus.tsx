import { SelectProps } from 'antd';
import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { StaticEnumDictionary } from '../../../../hooks/useStaticStatus/index.type';
import {
  GetSqlManageListFilterAuditLevelEnum,
  GetSqlManageListFilterSourceEnum,
  GetSqlManageListFilterStatusEnum,
} from '../../../../api/SqlManage/index.enum';

export const sourceDictionary: StaticEnumDictionary<GetSqlManageListFilterSourceEnum> =
  {
    [GetSqlManageListFilterSourceEnum.sql_audit_record]:
      'sqlManagement.selectDictionary.apiAudit',
    [GetSqlManageListFilterSourceEnum.audit_plan]:
      'sqlManagement.selectDictionary.auditPlan',
  };

export const auditLevelDictionary: StaticEnumDictionary<GetSqlManageListFilterAuditLevelEnum> =
  {
    [GetSqlManageListFilterAuditLevelEnum.normal]:
      'sqlManagement.selectDictionary.normal',
    [GetSqlManageListFilterAuditLevelEnum.notice]:
      'sqlManagement.selectDictionary.notice',
    [GetSqlManageListFilterAuditLevelEnum.warn]:
      'sqlManagement.selectDictionary.warn',
    [GetSqlManageListFilterAuditLevelEnum.error]:
      'sqlManagement.selectDictionary.error',
  };

export const statusDictionary: StaticEnumDictionary<GetSqlManageListFilterStatusEnum> =
  {
    [GetSqlManageListFilterStatusEnum.unhandled]:
      'sqlManagement.selectDictionary.unhandled',
    [GetSqlManageListFilterStatusEnum.solved]:
      'sqlManagement.selectDictionary.solved',
    [GetSqlManageListFilterStatusEnum.ignored]:
      'sqlManagement.selectDictionary.ignored',
  };

const useStaticStatus = () => {
  const { t } = useTranslation();

  const generateSourceSelectOptions: SelectProps['options'] = useMemo(() => {
    return Object.keys(sourceDictionary).map((key) => ({
      label: t(
        sourceDictionary[key as keyof typeof GetSqlManageListFilterSourceEnum]
      ),
      value: key,
    }));
  }, [t]);

  const generateAuditLevelSelectOptions: SelectProps['options'] =
    useMemo(() => {
      return Object.keys(auditLevelDictionary).map((key) => ({
        label: t(
          auditLevelDictionary[
            key as keyof typeof GetSqlManageListFilterAuditLevelEnum
          ]
        ),
        value: key,
      }));
    }, [t]);

  const generateStatusSelectOptions: SelectProps['options'] = useMemo(() => {
    return Object.keys(statusDictionary).map((key) => ({
      label: t(
        statusDictionary[key as keyof typeof GetSqlManageListFilterStatusEnum]
      ),
      value: key,
    }));
  }, [t]);
  return {
    generateSourceSelectOptions,
    generateAuditLevelSelectOptions,
    generateStatusSelectOptions,
  };
};

export default useStaticStatus;
