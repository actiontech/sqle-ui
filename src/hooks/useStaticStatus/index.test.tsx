import { renderHook } from '@testing-library/react-hooks';
import {
  render,
  fireEvent,
  screen,
  act as reactAct,
} from '@testing-library/react';

import { Select } from 'antd';
import useStaticStatus from '.';

describe('useRuleTemplate', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  test('should render static option for audit status', async () => {
    const { result } = renderHook(() => useStaticStatus());

    const { baseElement: baseElementWithOptions } = render(
      <Select data-testid="testId" value="value1">
        {result.current.generateAuditStatusSelectOption()}
      </Select>
    );
    expect(baseElementWithOptions).toMatchSnapshot();

    reactAct(() => {
      fireEvent.mouseDown(screen.getByText('value1'));
      jest.runAllTimers();
    });

    await screen.findAllByText('audit.auditStatus.initialized');
    expect(baseElementWithOptions).toMatchSnapshot();
  });

  test('should render static option for exec status', async () => {
    const { result } = renderHook(() => useStaticStatus());

    const { baseElement: baseElementWithOptions } = render(
      <Select data-testid="testId" value="value1">
        {result.current.generateExecStatusSelectOption()}
      </Select>
    );
    expect(baseElementWithOptions).toMatchSnapshot();

    reactAct(() => {
      fireEvent.mouseDown(screen.getByText('value1'));
      jest.runAllTimers();
    });

    await screen.findAllByText('audit.execStatus.initialized');
    expect(baseElementWithOptions).toMatchSnapshot();
  });

  test('should render static option for order status', async () => {
    const { result } = renderHook(() => useStaticStatus());

    const { baseElement: baseElementWithOptions } = render(
      <Select data-testid="testId" value="value1">
        {result.current.generateOrderStatusSelectOption()}
      </Select>
    );
    expect(baseElementWithOptions).toMatchSnapshot();

    reactAct(() => {
      fireEvent.mouseDown(screen.getByText('value1'));
      jest.runAllTimers();
    });

    await screen.findAllByText('order.status.process');
    expect(baseElementWithOptions).toMatchSnapshot();
  });

  test('should render static option for sql task status', async () => {
    const { result } = renderHook(() => useStaticStatus());

    const { baseElement: baseElementWithOptions } = render(
      <Select data-testid="testId" value="value1">
        {result.current.generateSqlTaskStatusSelectOption()}
      </Select>
    );
    expect(baseElementWithOptions).toMatchSnapshot();

    reactAct(() => {
      fireEvent.mouseDown(screen.getByText('value1'));
      jest.runAllTimers();
    });

    await screen.findAllByText('order.sqlTaskStatus.initialized');
    expect(baseElementWithOptions).toMatchSnapshot();
  });

  test('should render static option for workflow step type', async () => {
    const { result } = renderHook(() => useStaticStatus());

    const { baseElement: baseElementWithOptions } = render(
      <Select data-testid="testId" value="value1">
        {result.current.generateWorkflowStepTypeSelectOption()}
      </Select>
    );
    expect(baseElementWithOptions).toMatchSnapshot();

    reactAct(() => {
      fireEvent.mouseDown(screen.getByText('value1'));
      jest.runAllTimers();
    });

    await screen.findAllByText('order.workflowStatus.review');
    expect(baseElementWithOptions).toMatchSnapshot();
  });
});
