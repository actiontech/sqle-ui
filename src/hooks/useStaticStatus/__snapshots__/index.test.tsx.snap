// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRuleTemplate should render static option for audit status 1`] = `
<body>
  <div>
    <div
      class="ant-select ant-select-single ant-select-show-arrow"
      data-testid="testId"
    >
      <div
        class="ant-select-selector"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title="value1"
        >
          value1
        </span>
      </div>
      <span
        aria-hidden="true"
        class="ant-select-arrow"
        style="user-select: none;"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`useRuleTemplate should render static option for audit status 2`] = `
<body>
  <div>
    <div
      class="ant-select ant-select-single ant-select-show-arrow ant-select-open"
      data-testid="testId"
    >
      <div
        class="ant-select-selector"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-activedescendant="rc_select_TEST_OR_SSR_list_-1"
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="true"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title="value1"
        >
          value1
        </span>
      </div>
      <span
        aria-hidden="true"
        class="ant-select-arrow"
        style="user-select: none;"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="ant-select-dropdown"
        style="opacity: 0; pointer-events: none; min-width: 0; width: 0px;"
      >
        <div>
          <div
            id="rc_select_TEST_OR_SSR_list"
            role="listbox"
            style="height: 0px; width: 0px; overflow: hidden;"
          >
            <div
              aria-label="audit.auditStatus.initialized"
              aria-selected="false"
              id="rc_select_TEST_OR_SSR_list_0"
              role="option"
            >
              initialized
            </div>
          </div>
          <div
            class="rc-virtual-list"
            style="position: relative;"
          >
            <div
              class="rc-virtual-list-holder"
              style="max-height: 256px; overflow-y: hidden;"
            >
              <div>
                <div
                  class="rc-virtual-list-holder-inner"
                  style="display: flex; flex-direction: column;"
                >
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="audit.auditStatus.initialized"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      audit.auditStatus.initialized
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="audit.auditStatus.doing"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      audit.auditStatus.doing
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="audit.auditStatus.finished"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      audit.auditStatus.finished
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rc-virtual-list-scrollbar"
              style="width: 8px; top: 0px; bottom: 0px; right: 0px; position: absolute; display: none;"
            >
              <div
                class="rc-virtual-list-scrollbar-thumb"
                style="width: 100%; height: 128px; top: 0px; left: 0px; position: absolute; background: rgba(0, 0, 0, 0.5); border-radius: 99px; cursor: pointer; user-select: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`useRuleTemplate should render static option for exec status 1`] = `
<body>
  <div>
    <div
      class="ant-select ant-select-single ant-select-show-arrow"
      data-testid="testId"
    >
      <div
        class="ant-select-selector"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title="value1"
        >
          value1
        </span>
      </div>
      <span
        aria-hidden="true"
        class="ant-select-arrow"
        style="user-select: none;"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`useRuleTemplate should render static option for exec status 2`] = `
<body>
  <div>
    <div
      class="ant-select ant-select-single ant-select-show-arrow ant-select-open"
      data-testid="testId"
    >
      <div
        class="ant-select-selector"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-activedescendant="rc_select_TEST_OR_SSR_list_-1"
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="true"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title="value1"
        >
          value1
        </span>
      </div>
      <span
        aria-hidden="true"
        class="ant-select-arrow"
        style="user-select: none;"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="ant-select-dropdown"
        style="opacity: 0; pointer-events: none; min-width: 0; width: 0px;"
      >
        <div>
          <div
            id="rc_select_TEST_OR_SSR_list"
            role="listbox"
            style="height: 0px; width: 0px; overflow: hidden;"
          >
            <div
              aria-label="audit.execStatus.initialized"
              aria-selected="false"
              id="rc_select_TEST_OR_SSR_list_0"
              role="option"
            >
              initialized
            </div>
          </div>
          <div
            class="rc-virtual-list"
            style="position: relative;"
          >
            <div
              class="rc-virtual-list-holder"
              style="max-height: 256px; overflow-y: hidden;"
            >
              <div>
                <div
                  class="rc-virtual-list-holder-inner"
                  style="display: flex; flex-direction: column;"
                >
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="audit.execStatus.initialized"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      audit.execStatus.initialized
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="audit.execStatus.doing"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      audit.execStatus.doing
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="audit.execStatus.succeeded"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      audit.execStatus.succeeded
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="audit.execStatus.failed"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      audit.execStatus.failed
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rc-virtual-list-scrollbar"
              style="width: 8px; top: 0px; bottom: 0px; right: 0px; position: absolute; display: none;"
            >
              <div
                class="rc-virtual-list-scrollbar-thumb"
                style="width: 100%; height: 128px; top: 0px; left: 0px; position: absolute; background: rgba(0, 0, 0, 0.5); border-radius: 99px; cursor: pointer; user-select: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`useRuleTemplate should render static option for order status 1`] = `
<body>
  <div>
    <div
      class="ant-select ant-select-single ant-select-show-arrow"
      data-testid="testId"
    >
      <div
        class="ant-select-selector"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title="value1"
        >
          value1
        </span>
      </div>
      <span
        aria-hidden="true"
        class="ant-select-arrow"
        style="user-select: none;"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`useRuleTemplate should render static option for order status 2`] = `
<body>
  <div>
    <div
      class="ant-select ant-select-single ant-select-show-arrow ant-select-open"
      data-testid="testId"
    >
      <div
        class="ant-select-selector"
      >
        <span
          class="ant-select-selection-search"
        >
          <input
            aria-activedescendant="rc_select_TEST_OR_SSR_list_-1"
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded="true"
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autocomplete="off"
            class="ant-select-selection-search-input"
            id="rc_select_TEST_OR_SSR"
            readonly=""
            role="combobox"
            style="opacity: 0;"
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          class="ant-select-selection-item"
          title="value1"
        >
          value1
        </span>
      </div>
      <span
        aria-hidden="true"
        class="ant-select-arrow"
        style="user-select: none;"
        unselectable="on"
      >
        <span
          aria-label="down"
          class="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="ant-select-dropdown"
        style="opacity: 0; pointer-events: none; min-width: 0; width: 0px;"
      >
        <div>
          <div
            id="rc_select_TEST_OR_SSR_list"
            role="listbox"
            style="height: 0px; width: 0px; overflow: hidden;"
          >
            <div
              aria-label="order.status.wait_for_audit"
              aria-selected="false"
              id="rc_select_TEST_OR_SSR_list_0"
              role="option"
            >
              wait_for_audit
            </div>
          </div>
          <div
            class="rc-virtual-list"
            style="position: relative;"
          >
            <div
              class="rc-virtual-list-holder"
              style="max-height: 256px; overflow-y: hidden;"
            >
              <div>
                <div
                  class="rc-virtual-list-holder-inner"
                  style="display: flex; flex-direction: column;"
                >
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="order.status.wait_for_audit"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      order.status.wait_for_audit
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="order.status.wait_for_execution"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      order.status.wait_for_execution
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="order.status.executing"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      order.status.executing
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="order.status.finished"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      order.status.finished
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="order.status.reject"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      order.status.reject
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="order.status.exec_failed"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      order.status.exec_failed
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                  <div
                    aria-selected="false"
                    class="ant-select-item ant-select-item-option"
                    title="order.status.canceled"
                  >
                    <div
                      class="ant-select-item-option-content"
                    >
                      order.status.canceled
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-item-option-state"
                      style="user-select: none;"
                      unselectable="on"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rc-virtual-list-scrollbar"
              style="width: 8px; top: 0px; bottom: 0px; right: 0px; position: absolute; display: none;"
            >
              <div
                class="rc-virtual-list-scrollbar-thumb"
                style="width: 100%; height: 128px; top: 0px; left: 0px; position: absolute; background: rgba(0, 0, 0, 0.5); border-radius: 99px; cursor: pointer; user-select: none;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
